src/python/gudhi/representations/vector_methods.py

```diff
@@ -104,10 +104,11 @@ def fit(self, X, y=None):
             X (list of n x 2 numpy arrays): input persistence diagrams.
             y (n x 1 array): persistence diagram labels (unused).
         """
+        self.nan_in_range = np.isnan(np.array(self.sample_range))
         if np.isnan(np.array(self.sample_range)).any():
```

Comment:
Shouldn't this use the new variable defined on the line just above?

Reply:
Indeed. Should be fixed now.